### PR TITLE
Adding test cases for gadget usage tracking

### PR DESCRIPTION
**Pull request by Megala21**

### CODE REVIEW

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        String[][] gadgetMapping = { { "publisher", "b" }, { "subscriber", "d" } };
+        script = generateAddGadgetScript(gadgetMapping);
+        getDriver().navigate().refresh();
+        selectPane("gadgets");
+        Thread.sleep(2000);
+        getDriver().executeScript(script);
+        Thread.sleep(2000);
+        redirectToLocation(DS_HOME_CONTEXT, DS_DASHBOARDS_CONTEXT);
+    }
+
```
Comment (NipunaMarcus) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: No need to leave unnecessary spaces. Please remove them as they won't help increase the readability.

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+     * To create some gadgets in 2 dashboards
+     *
+     * @throws MalformedURLException
+     * @throws XPathExpressionException
+     * @throws InterruptedException
+     */
+    private void createRelevantGadgets() throws MalformedURLException, XPathExpressionException, InterruptedException {
+        redirectToLocation(DS_HOME_CONTEXT, DS_DASHBOARDS_CONTEXT);
+        getDriver().findElement(By.cssSelector("#" + DASHBOARD1_TITLE + " a.ues-edit")).click();
+        String[][] gadgetMappings = { { "publisher", "b" }, { "usa-map", "c" } };
```
Comment (NipunaMarcus) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: No need to leave these unnecessary spaces when declaring a map. Please use a code formatting tool which will help in this case.

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        String script = generateAddGadgetScript(gadgetMappings);
+        getDriver().navigate().refresh();
+        selectPane("gadgets");
+        Thread.sleep(2000);
+        getDriver().executeScript(script);
+        Thread.sleep(2000);
+        redirectToLocation(DS_HOME_CONTEXT, DS_DASHBOARDS_CONTEXT);
+        getDriver().findElement(By.cssSelector("#" + DASHBOARD2_TITLE + " a.ues-edit")).click();
+        String[][] gadgetMapping = { { "publisher", "b" }, { "subscriber", "d" } };
```
Comment (NipunaMarcus) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: No need to leave these unnecessary spaces when declaring a map. Please use a code formatting tool which will help in this case.

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    }
+
+    /**
+     * Provides user modes.
+     *
+     * @return user modes
+     */
+    @DataProvider(name = "userMode")
+    public static Object[][] userModeProvider() {
+        return new Object[][] { { TestUserMode.SUPER_TENANT_ADMIN } };
```
Comment (NipunaMarcus) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: No need to leave these unnecessary spaces when declaring a array. Please use a code formatting tool which will help in this case.

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java (hunk 1 above)
Comment (Megala21) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: Fixed with #133

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java (hunk 2 above)
Comment (Megala21) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: This was formatted using wso2 code formatting template for eclipse. So I hope this ok.

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java (hunk 3 above)
Comment (Megala21) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: This was formatted using wso2 code formatting template for eclipse. So I hope this ok.

File: modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java (hunk 4 above)
Comment (Megala21) [inline modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java]: This was formatted using wso2 code formatting template for eclipse. So I hope these spaces are acceptable.

File: modules/integration/tests-ui-integration/tests-ui/src/test/resources/files/portal.json [hunk 5]
```diff
@@ -4,17 +4,25 @@
...
   "authentication": {
     "activeMethod": "basic",
     "methods": {
       "sso": {
         "attributes": {
           "issuer": "portal",
-          "identityProviderURL": "%https.ip%/samlsso",
-          "responseSigningEnabled": "false",
-          "acs": "%https.ip%/portal/acs",
+          "identityProviderURL": "https://localhost:9443/samlsso",
```
Comment (sinthuja) [inline modules/integration/tests-ui-integration/tests-ui/src/test/resources/files/portal.json]: Needs to be parameterized as above, why this change is needed?

Comment (Megala21) [inline modules/integration/tests-ui-integration/tests-ui/src/test/resources/files/portal.json]: I updated the portal.json in consistent with the portal.json with carbon-dashboards. This was changed with https://github.com/wso2/carbon-dashboards/commit/2fbe8838a04015b0584ac4ae470a7056c10d230f#diff-b1d0f8773ac16551d7910e5033f7bc0b
